src/main/kotlin/uk/co/craigbass/pratura/boundary/Pratura.kt

```diff
@@ -20,10 +23,15 @@ abstract class Pratura : Application() {
       )
       ViewBasket::class -> uk.co.craigbass.pratura.usecase.ViewBasket(
         basketItemsRetriever,
-        productRetriever
+        productRetriever,
+        currencyRetriever
       )
       AddProduct::class -> uk.co.craigbass.pratura.usecase.AddProduct(productSaver)
-      ViewAllProducts::class -> uk.co.craigbass.pratura.usecase.ViewAllProducts(productRetriever)
+      ViewAllProducts::class -> uk.co.craigbass.pratura.usecase.ViewAllProducts(
```

Comment:
could you let the request type imply the use case
like is there ever not a 1:1 mapping?

Then you wouldn't always have to supply the class name of the use case and your coupling would be even looser